File: algorithms/search/binary_search/binary_search.go
package main

import "fmt"

func search(datalist []int, key int) int {
	start := 0
	end := len(datalist) - 1
	for start <= end {
		mid := start + (end-start)/2
		if datalist[mid] == key {
			return mid
		}
		if datalist[mid] < key {
			start = mid + 1
		} else {
			end = mid - 1
		}
	}
	return -1
}

func result(idx int) {
	if idx == -1 {
		fmt.Println("Element not found")
	} else {
		fmt.Println("Element found at index", idx)
	}
}

func main() {
	arr := []int{-5, 0, 1, 2, 9, 17, 42, 69}
	result(search(arr, 3))
	result(search(arr, 2))
}

File: algorithms/search/binary_search/binary_search.py
def search(arr, key):
    start = 0
    end = len(arr) - 1

    while start <= end:
        mid = start + (end - start)//2
        if arr[mid] == key:
            return mid
        elif arr[mid] < key:
            start = mid + 1
        else:
            end = mid - 1
  
    return -1

def result(idx):
    if idx == -1:
        print("Element not found")
    else:
        print("Element found at index", idx)

if __name__ == "__main__":
    arr = [-5, 0, 1, 2, 9, 17, 42, 69]
    result(search(arr, 3))
    result(search(arr, 2))